llvm/include/llvm/ADT/GenericCycleInfo.h

```diff
@@ -189,6 +195,21 @@ template <typename ContextT> class GenericCycle {
   //@{
   using const_entry_iterator =
       typename SmallVectorImpl<BlockT *>::const_iterator;
+  const_entry_iterator entry_begin() const {
+    return const_entry_iterator{Entries.begin()};
```

Comment:
nit: you should be able to drop the redundant return types here and below
```suggestion
    return {Entries.begin()};
```

Reply:
Thanks! Fixed in an intermediate patch.

Reply:
Fixed.